source/MRMesh/MRVoxelsVolumeAccess.h

```diff
@@ -188,9 +133,12 @@ class VoxelsVolumeCachingAccessor
         : accessor_( accessor )
         , indexer_( indexer )
         , params_( std::move( parameters ) )
-        , layers_( params_.preloadedLayerCount, std::vector<ValueType>( indexer_.sizeXY() ) )
+        , layers_( params_.preloadedLayerCount )
+        , firstLayerVoxelId_( params_.preloadedLayerCount )
     {
         assert( params_.preloadedLayerCount > 0 );
+        for ( auto & l : layers_ )
```

Comment:
Does this variant have any advantages?

Reply:
Original version first created a temporary array, then copied it in all layers, and then deleted a temporary. New version just allocates necessary memory and initializes it with zeros.